main/autogen/autogen.cc

```diff
@@ -326,13 +325,13 @@ class AutogenWalk {
 
 // Convert a Sorbet `ParsedFile` into an Autogen `ParsedFile` by walking it as above and also recording the checksum of
 // the current file
-ParsedFile Autogen::generate(core::Context ctx, ast::ParsedFile tree) {
+ParsedFile Autogen::generate(core::Context ctx, ast::ParsedFile tree, const CRCBuilder &crcBuilder) {
     AutogenWalk walk;
     tree.tree = ast::TreeMap::apply(ctx, walk, move(tree.tree));
     auto pf = walk.parsedFile();
     pf.path = string(tree.file.data(ctx).path());
     auto src = tree.file.data(ctx).source();
-    pf.cksum = CRC::Calculate(src.data(), src.size(), CRC::CRC_32());
+    pf.cksum = crcBuilder.crc32(src);
```

Comment:
Heh, nice find.  Is crc computation hot enough that we could consider using the crc instructions (SSE4.2) on x86-64?

Reply:
It's ~30% of this function's (Autogen::generate) runtime according to perf, so yeah! Do we have that instruction on devboxen?

Reply:
We do!  https://github.com/komrad36/CRC#option-11-1-byte-hardware-accelerated and following has a nice (if long) explanation of how to best use the instruction.  (Search for "OPTION 13" if you just want the gofast version and don't want any explanation of what's going on.)

Reply:
That's great. Seems like we could expect a ~10X speedup.

Reply:
Looks like we assume at least sandybridge, so this could be the default CRC calculation method w/ no fallback.